lib/galaxy/jobs/output_checker.py

```diff
@@ -146,10 +146,12 @@ def check_output(stdio_regexes, stdio_exit_codes, stdout, stderr, tool_exit_code
             if stderr:
                 state = DETECTED_JOB_STATE.GENERIC_ERROR
 
-        if DETECTED_JOB_STATE != DETECTED_JOB_STATE.OK and stderr:
+        if state != DETECTED_JOB_STATE.OK and stderr:
             if stderr:
```

Comment:
This check doesn't make sense to me, since it's already checked in the outer `if`.

Reply:
true. needs a fix